src/fhirtypes/StructureDefinition.ts

```diff
@@ -366,6 +367,10 @@ export class StructureDefinition {
    * @returns {ElementDefinition} the new ElementDefinition
    */
   newElement(name = '$UNKNOWN'): ElementDefinition {
+    //Check if there already exists an element that is defined by an ancestor
+    if (this.elements.find(e => e.id == `${this.id}.${name}`)) {
+      throw new ElementAreadyDefinedError(name);
```

Comment:
When I saw this messaged logged, I thought it would be helpful to include the definition that it is a part of. If we did that, the error message could look something like this:

```
Cannot define element extension on MyResource because it has already been defined
```

After looking around at a few other errors thrown in this file, I can't find many that actually include the definition name, so I could be wrong, but when I read the error message without looking at other errors, I thought this would be helpful.